demos/dialog.html

```diff
@@ -0,0 +1,147 @@
+            <li class="mdc-dialog__body__list-item">Callisto</li>
+            <li class="mdc-dialog__body__list-item">Ganymede</li>
+            <li class="mdc-dialog__body__list-item">Luna</li>
+            <li class="mdc-dialog__body__list-item">Marimba</li>
+            <li class="mdc-dialog__body__list-item">Schwifty</li>
+            <li class="mdc-dialog__body__list-item">Callisto</li>
+            <li class="mdc-dialog__body__list-item">Ganymede</li>
+            <li class="mdc-dialog__body__list-item">Luna</li>
+            <li class="mdc-dialog__body__list-item">Marimba</li>
+            <li class="mdc-dialog__body__list-item">Schwifty</li>
+          </ul>
+        </section>
+        <footer class="mdc-dialog__footer">
+          <button type="button" class="mdc-button mdc-dialog__footer__button mdc-dialog--cancel">DECLINE</button>
+          <button type="button" class="mdc-button mdc-dialog__footer__button mdc-dialog--accept">ACCEPT</button>
+        </footer>
+      </div>
+      <div class="mdc-dialog__backdrop"></div> 
+    </aside>
+    </div>
+    <section class="user-content">
+      <div>
+        <h2>MDC Web Dialog</h2>
+        <p>Modal dialog windows for the web</p>
+      </div>
+      <button id="dialog-activation" class="mdc-button mdc-button--primary mdc-button--raised">Show Dialog</button>
+      <button id="dialog-activation--scrollable" class="mdc-button mdc-button--primary mdc-button--raised">Show Scrolling Dialog</button>
+
+			<div class="mdc-form-field">
+        <div id="mdc-js-checkbox" class="mdc-checkbox mdc-checkbox--upgraded mdc-ripple-upgraded mdc-ripple-upgraded--unbounded mdc-ripple-upgraded--foreground-unbounded-activation" style="--mdc-ripple-surface-width:40px; --mdc-ripple-surface-height:40px; --mdc-ripple-fg-size:24px; --mdc-ripple-fg-unbounded-transform-duration:254.967ms; --mdc-ripple-fg-scale:2.77369; --mdc-ripple-left:8px; --mdc-ripple-top:8px;">
+          <input type="checkbox" id="toggle-rtl" class="mdc-checkbox__native-control">
+          <div class="mdc-checkbox__background">
+            <svg version="1.1" class="mdc-checkbox__checkmark" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" xml:space="preserve">
+              <path class="mdc-checkbox__checkmark__path" fill="none" stroke="white" d="M1.73,12.91 8.1,19.28 22.79,4.59"></path>
+            </svg>
+            <div class="mdc-checkbox__mixedmark"></div>
+          </div>
+        </div>
+        <label for="native-js-checkbox" id="my-checkbox-label">Make RTL</label>
+      </div> 
+    </section>
+
+    <script src="assets/material-components-web.js" charset="utf-8"></script>
+    <script>
+      var dialog = new mdc.dialog.MDCDialog(document.querySelector('#mdc-dialog-one'));
+      var dialogScrollable = new mdc.dialog.MDCDialog(document.querySelector('#mdc-dialog-two'));
+      document.querySelector('#dialog-activation').addEventListener('click', function (evt) {
+        dialog.lastFocusedTarget = evt.target;
```

Comment:
Have you tried using [FocusEvent.relatedTarget](https://w3c.github.io/uievents/#idl-focusevent) to derive what the last focused target was, rather than make the user pass it in themselves?